### PR TITLE
chore: upgrade vault provider version to 4.0

### PR DESCRIPTION
**Pull request by ansgarm**
### Related issue

Fixes # 

### Description

In plain English, describe your approach to addressing the issue linked above. For example, if you made a particular design decision, let us know why you chose this path instead of another solution.

### Checklist

- [ ] I have updated the PR title to match [CDKTF's style guide](https://github.com/cdktf/.github/blob/main/CONTRIBUTING.md#pull-requests-1)
- [ ] I have run the linter on my code locally
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if applicable
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works if applicable
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by team-tf-cdk** [comment]
#### [`custom-constructs`] Terraform Plan 📖`success`

<a href="https://app.terraform.io/app/cdktf-team/custom-constructs/runs/run-nePYe8XeWnfqbemo" target="_blank">Remote Plan Link ↗️</a>

<details><summary>Show Plan</summary>

```tf
Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/cdktf-team/custom-constructs/runs/run-nePYe8XeWnfqbemo

Waiting for the plan to start...

Terraform v1.7.5
on linux_amd64
Initializing plugins and modules...
data.github_repository.cdktf-construct-cdktf-tf-module-stack_repo_1C630C5A: Reading...
data.github_repository.cdktf-construct-cdktf-cdk8s_repo_6A622B5E: Reading...
data.github_repository.cdktf-construct-cdktf-multi-stack-tfe_repo_5BB71C62: Reading...
data.github_team.cdktf-team-cdktf: Reading...
data.github_repository.cdktf-construct-cdktf-local-exec_repo_64EDC49D: Reading...
github_repository.cdktf-tf-module-stack-go_repo_A899C4D5: Refreshing state... [id=cdktf-tf-module-stack-go]
data.github_repository.cdktf-construct-cdktf-cdk8s_repo_6A622B5E: Read complete after 0s [id=cdktf-cdk8s]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-gh-token-alias-PROJEN_GITHUB_TOKEN_99DCA853: Refreshing state... [id=cdktf-cdk8s:PROJEN_GITHUB_TOKEN]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-gh-token-alias-GO_GITHUB_TOKEN_E3BEFE91: Refreshing state... [id=cdktf-cdk8s:GO_GITHUB_TOKEN]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-npm-token_C93B326E: Refreshing state... [id=cdktf-cdk8s:NPM_TOKEN]
github_issue_label.cdktf-construct-cdktf-cdk8s_repository-setup_auto-approve-label_46AAE156: Refreshing state... [id=cdktf-cdk8s:auto-approve]
github_branch_protection.cdktf-construct-cdktf-cdk8s_repository-setup_main-protection_8B8D1D76: Refreshing state... [id=BPR_kwDOGqHliM4CqkZi]
data.github_repository.cdktf-construct-cdktf-tf-module-stack_repo_1C630C5A: Read complete after 0s [id=cdktf-tf-module-stack]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-twine-password_A6335EE5: Refreshing state... [id=cdktf-cdk8s:TWINE_PASSWORD]
data.github_repository.cdktf-construct-cdktf-local-exec_repo_64EDC49D: Read complete after 0s [id=cdktf-local-exec]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-twine-username_EF6824FB: Refreshing state... [id=cdktf-cdk8s:TWINE_USERNAME]
data.github_repository.cdktf-construct-cdktf-multi-stack-tfe_repo_5BB71C62: Read complete after 0s [id=cdktf-multi-stack-tfe]
github_issue_label.cdktf-construct-cdktf-cdk8s_repository-setup_automerge-label_A5C2F715: Refreshing state... [id=cdktf-cdk8s:automerge]
github_actions_secret.cdktf-construct-cdktf-cdk8s_repo_secret-gh-token_FDFB28D6: Refreshing state... [id=cdktf-cdk8s:GH_TOKEN]
github_repository_webhook.cdktf-construct-cdktf-cdk8s_repository-setup_slack-webhook_05B55983: Refreshing state... [id=385811429]
github_issue_label.cdktf-construct-cdktf-cdk8s_repository-setup_no-auto-close-label_16547785: Refreshing state... [id=cdktf-cdk8s:no-auto-close]
github_repository_webhook.cdktf-construct-cdktf-tf-module-stack_repository-setup_slack-webhook_1546DD10: Refreshing state... [id=385811493]
github_branch_protection.cdktf-construct-cdktf-tf-module-stack_repository-setup_main-protection_1387AE3A: Refreshing state... [id=BPR_kwDOHBgud84CqkZj]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-nuget-api-key_251EB843: Refreshing state... [id=cdktf-tf-module-stack:NUGET_API_KEY]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-gh-token_5CC4BD3B: Refreshing state... [id=cdktf-tf-module-stack:GH_TOKEN]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-maven-staging-profile-id_4AE06017: Refreshing state... [id=cdktf-tf-module-stack:MAVEN_STAGING_PROFILE_ID]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-twine-password_02ABFB12: Refreshing state... [id=cdktf-tf-module-stack:TWINE_PASSWORD]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-maven-username_74367FC1: Refreshing state... [id=cdktf-tf-module-stack:MAVEN_USERNAME]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-gh-token-alias-GO_GITHUB_TOKEN_8A129DF1: Refreshing state... [id=cdktf-tf-module-stack:GO_GITHUB_TOKEN]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-maven-password_5AED5C2F: Refreshing state... [id=cdktf-tf-module-stack:MAVEN_PASSWORD]
github_issue_label.cdktf-construct-cdktf-tf-module-stack_repository-setup_no-auto-close-label_174597DF: Refreshing state... [id=cdktf-tf-module-stack:no-auto-close]
github_issue_label.cdktf-construct-cdktf-tf-module-stack_repository-setup_automerge-label_01B3DA9F: Refreshing state... [id=cdktf-tf-module-stack:automerge]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-npm-token_4A60F3F6: Refreshing state... [id=cdktf-tf-module-stack:NPM_TOKEN]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-maven-gpg-private-key-passphrase_9F164680: Refreshing state... [id=cdktf-tf-module-stack:MAVEN_GPG_PRIVATE_KEY_PASSPHRASE]
github_issue_label.cdktf-construct-cdktf-tf-module-stack_repository-setup_auto-approve-label_5CF06293: Refreshing state... [id=cdktf-tf-module-stack:auto-approve]
data.github_team.cdktf-team-cdktf: Read complete after 6s [id=6602018]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-gh-token-alias-PROJEN_GITHUB_TOKEN_F74237D9: Refreshing state... [id=cdktf-tf-module-stack:PROJEN_GITHUB_TOKEN]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-twine-username_E2283133: Refreshing state... [id=cdktf-tf-module-stack:TWINE_USERNAME]
github_actions_secret.cdktf-construct-cdktf-tf-module-stack_repo_secret-maven-gpg-private-key_DFA54479: Refreshing state... [id=cdktf-tf-module-stack:MAVEN_GPG_PRIVATE_KEY]
github_issue_label.cdktf-construct-cdktf-local-exec_repository-setup_no-auto-close-label_ED807439: Refreshing state... [id=cdktf-local-exec:no-auto-close]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-gh-token-alias-GO_GITHUB_TOKEN_D8006675: Refreshing state... [id=cdktf-local-exec:GO_GITHUB_TOKEN]
github_issue_label.cdktf-construct-cdktf-local-exec_repository-setup_auto-approve-label_58C7EDEC: Refreshing state... [id=cdktf-local-exec:auto-approve]
github_issue_label.cdktf-construct-cdktf-local-exec_repository-setup_automerge-label_A8F9C80E: Refreshing state... [id=cdktf-local-exec:automerge]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-gh-token-alias-PROJEN_GITHUB_TOKEN_673FEB36: Refreshing state... [id=cdktf-local-exec:PROJEN_GITHUB_TOKEN]
github_branch_protection.cdktf-construct-cdktf-local-exec_repository-setup_main-protection_356B7C90: Refreshing state... [id=BPR_kwDOGmCyjs4CqkZl]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-npm-token_668831D4: Refreshing state... [id=cdktf-local-exec:NPM_TOKEN]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-twine-password_92687429: Refreshing state... [id=cdktf-local-exec:TWINE_PASSWORD]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-twine-username_740C2F06: Refreshing state... [id=cdktf-local-exec:TWINE_USERNAME]
github_actions_secret.cdktf-construct-cdktf-local-exec_repo_secret-gh-token_83854260: Refreshing state... [id=cdktf-local-exec:GH_TOKEN]
github_repository_webhook.cdktf-construct-cdktf-local-exec_repository-setup_slack-webhook_1BC61C08: Refreshing state... [id=385811432]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-gh-token_37CD874B: Refreshing state... [id=cdktf-multi-stack-tfe:GH_TOKEN]
github_issue_label.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_automerge-label_5383C30F: Refreshing state... [id=cdktf-multi-stack-tfe:automerge]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-twine-username_A2A982CF: Refreshing state... [id=cdktf-multi-stack-tfe:TWINE_USERNAME]
github_branch_protection.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_main-protection_3A5796DA: Refreshing state... [id=BPR_kwDOG0VkRs4CqkZk]
github_repository_webhook.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_slack-webhook_9CBC5F54: Refreshing state... [id=385811467]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-npm-token_FB3DBBDB: Refreshing state... [id=cdktf-multi-stack-tfe:NPM_TOKEN]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-gh-token-alias-PROJEN_GITHUB_TOKEN_C6829094: Refreshing state... [id=cdktf-multi-stack-tfe:PROJEN_GITHUB_TOKEN]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-gh-token-alias-GO_GITHUB_TOKEN_5BED1C4A: Refreshing state... [id=cdktf-multi-stack-tfe:GO_GITHUB_TOKEN]
github_issue_label.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_auto-approve-label_C36794CF: Refreshing state... [id=cdktf-multi-stack-tfe:auto-approve]
github_issue_label.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_no-auto-close-label_69A4EAA3: Refreshing state... [id=cdktf-multi-stack-tfe:no-auto-close]
github_actions_secret.cdktf-construct-cdktf-multi-stack-tfe_repo_secret-twine-password_5CADBB76: Refreshing state... [id=cdktf-multi-stack-tfe:TWINE_PASSWORD]
github_issue_label.cdktf-tf-module-stack-go_repository-setup_no-auto-close-label_57244BF2: Refreshing state... [id=cdktf-tf-module-stack-go:no-auto-close]
github_issue_label.cdktf-tf-module-stack-go_repository-setup_auto-approve-label_29FEBEFF: Refreshing state... [id=cdktf-tf-module-stack-go:auto-approve]
github_issue_label.cdktf-tf-module-stack-go_repository-setup_automerge-label_180DA270: Refreshing state... [id=cdktf-tf-module-stack-go:automerge]
github_repository_webhook.cdktf-tf-module-stack-go_repository-setup_slack-webhook_B1E661E0: Refreshing state... [id=387516719]
github_team_repository.cdktf-construct-cdktf-cdk8s_repository-setup_managing-team_105FF4C2: Refreshing state... [id=6602018:cdktf-cdk8s]
github_team_repository.cdktf-construct-cdktf-tf-module-stack_repository-setup_managing-team_823F55F7: Refreshing state... [id=6602018:cdktf-tf-module-stack]
github_team_repository.cdktf-tf-module-stack-go_repository-setup_managing-team_1A21B533: Refreshing state... [id=6602018:cdktf-tf-module-stack-go]
github_team_repository.cdktf-construct-cdktf-multi-stack-tfe_repository-setup_managing-team_01DE83DC: Refreshing state... [id=6602018:cdktf-multi-stack-tfe]
github_team_repository.cdktf-construct-cdktf-local-exec_repository-setup_managing-team_E9BA7A6F: Refreshing state... [id=6602018:cdktf-local-exec]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
╷
│ Warning: Value for undeclared variable
│ 
│ The root module does not declare a variable named
│ "alert-prs-slack-webhook-url" but a value was found in file
│ "/home/tfc-agent/.tfc-agent/component/terraform/runs/run-nePYe8XeWnfqbemo/terraform.tfvars".
│ If you meant to use this value, add a "variable" block to the
│ configuration.
│ 
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵

```

</details>

*Pusher: @ansgarm, Action: `pull_request_target`, Working Directory: ``, Workflow: `Plan`*

**Review by DanielMSchmidt — APPROVED** [review]
